### PR TITLE
[CIS-1187] Fix channel list showing `currentUser.mutedChannels` initially

### PR DESCRIPTION
**Pull request by evsaev**
This **PR** fixes the bug when all updated/inserted channels are added to the channel list before the underlaying query is synced. The fix updates `ChannelListController` to handle unlinked channels only when the `controller.synchronize` is successfully completed.

### CODE REVIEW

File: Sources/StreamChat/Database/DatabaseSession.swift [hunk 1]
```diff
@@ -167,6 +167,10 @@ protocol ChannelDatabaseSession {
         query: ChannelListQuery?
     ) throws -> ChannelDTO
     
+    /// Loads channel list query with the given filter hash from the database.
+    /// - Parameter filterHash: The filter hash.
+    func channelListQuery(filterHash: String) -> ChannelListQueryDTO?
+
```
Comment (nuno-vieira) [inline Sources/StreamChat/Database/DatabaseSession.swift:173]: Why was this added? 🤔 I don't see it being used

File: Sources/StreamChat/Controllers/ChannelListController/ChannelListController.swift [hunk 2]
```diff
@@ -241,27 +241,35 @@ public class ChatChannelListController: DataController, DelegateCallable, DataSt
...
+            }
+            
             for channel in channels {
                 guard let channelDTO = session.channel(cid: channel.cid) else {
                     log.error("Channel \(channel.cid) cannot be found in database.")
                     continue
                 }
-                let query = session.saveQuery(query: self.query)
-                query.channels.remove(channelDTO)
+                queryDTO.channels.insert(channelDTO)
```
Comment (nuno-vieira) [inline Sources/StreamChat/Controllers/ChannelListController/ChannelListController.swift:272]: Could we somehow add tests to make sure there are no regressions in the future?

Comment (b-onc) [reply Sources/StreamChat/Controllers/ChannelListController/ChannelListController.swift:272]: Used in the `guard` statements

**Review by b-onc — DISMISSED** [review]
Interesting find! This can use a test case, and CHANGELOG since it's released

File: Sources/StreamChat/Database/DatabaseSession.swift (hunk 1 above)
Comment (evsaev) [reply Sources/StreamChat/Database/DatabaseSession.swift:173]: Used in `ChannelListController.link/unlink` funcs 🙂

File: Sources/StreamChat/Controllers/ChannelListController/ChannelListController.swift (hunk 2 above)
Comment (evsaev) [reply Sources/StreamChat/Controllers/ChannelListController/ChannelListController.swift]: ✅ Added tests

**Comment by evsaev** [comment]
> Interesting find! This can use a test case, and CHANGELOG since it's released

✅ Added tests and CHANGELOG entry

File: Sources/StreamChat/Controllers/ChannelListController/ChannelListController_Tests.swift [hunk 3]
```diff
@@ -255,42 +255,28 @@ class ChannelListController_Tests: XCTestCase {
...
-            }
-            
-            func controller(_ controller: ChatChannelListController, shouldListUpdatedChannel channel: ChatChannel) -> Bool {
-                shouldListUpdatedChannel(channel)
-            }
-        }
-        
-        // Simulate `synchronize` call
-        controller.synchronize()
+    func test_newChannel_callsListHook_whenSynchronized() throws {
```
Comment (b-onc) [inline Sources/StreamChat/Controllers/ChannelListController/ChannelListController_Tests.swift:258]: We should make sure this hook is not called before `sync` is called

Comment (evsaev) [reply Sources/StreamChat/Controllers/ChannelListController/ChannelListController_Tests.swift:258]: There's a `test_unlinkedChannels_doNotTriggerHooks_whenNotSynchronized`. 

And we can handle updates of linked channel even when a query is not yet synced. It is covered by `test_linkedChannels_doesTriggerUnlinkHook_whenNotSynchronized`

**Review by b-onc — APPROVED** [review]

**Review by nuno-vieira — APPROVED** [review]
LGTM ✅

**Review by tbarbugli — CHANGES_REQUESTED** [review]
muted channels are not showed anymore but "mute channel" action from the demo app is not working (channel list, slide, mute channel)
